bookserver/routers/discuss.py

```diff
@@ -21,19 +70,162 @@
 templates = Jinja2Templates(directory=f"{settings._book_server_path}/templates/discuss")
 
 
-# .. _login:
-#
-# login
-# -----
-@router.get("/home.html", response_class=HTMLResponse)
-def login_form(request: Request):
-    return templates.TemplateResponse("home.html", {"request": request})
-
-
-@router.websocket_route("/ws")
-async def websocket_endpoint(websocket: WebSocket):
-    rslogger.debug("IN WEBSOCKET")
-    await websocket.accept()
-    while True:
-        data = await websocket.receive_text()
-        await websocket.send_text(f"Message sent was {data}")
+class ConnectionManager:
+    def __init__(self):
+        self.active_connections: Dict[str, WebSocket] = {}
+
+    async def connect(self, user: str, websocket: WebSocket):
+        await websocket.accept()
+        self.active_connections[user] = websocket
```

Comment:
Perhaps `assert user not in active_connections`?

Reply:
I think that could potentially make a race condition.  When a user refreshes the page the web socket closes and then reopens with a call (potentially) to the same server.    The server will also note this and remove the user from the active connections.  I think the assert just needlessly creates a new fail condition.

Reply:
OK, sounds good.